tests/src/unit-inspection.cpp

```diff
@@ -202,44 +202,55 @@ TEST_CASE("object inspection")
+        json const j
+        {
+            {"object", json::object()},
+            {"array", {1, 2, 3, 4}},
+            {"array2d", {{1, 2, 3}, {4, 5, 6}}},
+            {"array3d", {{{1, 2, 3}, {4, 5, 6}}, {{2, 3, 4}, {5, 6, 7}}}},
+            {"array4d", {{{{1, 2, 3}, {4, 5, 6}}, {{2, 3, 4}, {5, 6, 7}}}, {{{11, 12, 13}, {14, 15, 16}}, {{12, 13, 14}, {15, 16, 17}}}}},
+            {"number", 42},
+            {"boolean", false},
+            {"null", nullptr},
+            {"string", "Hello world"}
+        };
 
         SECTION("no indent / indent=-1")
         {
             CHECK(j.dump() ==
-                  "{\"array\":[1,2,3,4],\"boolean\":false,\"null\":null,\"number\":42,\"object\":{},\"string\":\"Hello world\"}");
+                  "{\"array\":[1,2,3,4],\"array2d\":[[1,2,3],[4,5,6]],\"array3d\":[[[1,2,3],[4,5,6]],[[2,3,4],[5,6,7]]],\"array4d\":[[[[1,2,3],[4,5,6]],[[2,3,4],[5,6,7]]],[[[11,12,13],[14,15,16]],[[12,13,14],[15,16,17]]]],\"boolean\":false,\"null\":null,\"number\":42,\"object\":{},\"string\":\"Hello world\"}");
 
             CHECK(j.dump() == j.dump(-1));
         }
 
         SECTION("indent=0")
         {
             CHECK(j.dump(0) ==
-                  "{\n\"array\": [\n1,\n2,\n3,\n4\n],\n\"boolean\": false,\n\"null\": null,\n\"number\": 42,\n\"object\": {},\n\"string\": \"Hello world\"\n}");
+                  "{\n\"array\": [1,2,3,4],\n\"array2d\": [\n[1,2,3],\n[4,5,6]\n],\n\"array3d\": [\n[\n[1,2,3],\n[4,5,6]\n],\n[\n[2,3,4],\n[5,6,7]\n]\n],\n\"array4d\": [\n[\n[\n[1,2,3],\n[4,5,6]\n],\n[\n[2,3,4],\n[5,6,7]\n]\n],\n[\n[\n[11,12,13],\n[14,15,16]\n],\n[\n[12,13,14],\n[15,16,17]\n]\n]\n],\n\"boolean\": false,\n\"null\": null,\n\"number\": 42,\n\"object\": {},\n\"string\": \"Hello world\"\n}");
         }
 
         SECTION("indent=1, space='\t'")
         {
             CHECK(j.dump(1, '\t') ==
-                  "{\n\t\"array\": [\n\t\t1,\n\t\t2,\n\t\t3,\n\t\t4\n\t],\n\t\"boolean\": false,\n\t\"null\": null,\n\t\"number\": 42,\n\t\"object\": {},\n\t\"string\": \"Hello world\"\n}");
+                  "{\n\t\"array\": [1,2,3,4],\n\t\"array2d\": [\n\t\t[1,2,3],\n\t\t[4,5,6]\n\t],\n\t\"array3d\": [\n\t\t[\n\t\t\t[1,2,3],\n\t\t\t[4,5,6]\n\t\t],\n\t\t[\n\t\t\t[2,3,4],\n\t\t\t[5,6,7]\n\t\t]\n\t],\n\t\"array4d\": [\n\t\t[\n\t\t\t[\n\t\t\t\t[1,2,3],\n\t\t\t\t[4,5,6]\n\t\t\t],\n\t\t\t[\n\t\t\t\t[2,3,4],\n\t\t\t\t[5,6,7]\n\t\t\t]\n\t\t],\n\t\t[\n\t\t\t[\n\t\t\t\t[11,12,13],\n\t\t\t\t[14,15,16]\n\t\t\t],\n\t\t\t[\n\t\t\t\t[12,13,14],\n\t\t\t\t[15,16,17]\n\t\t\t]\n\t\t]\n\t],\n\t\"boolean\": false,\n\t\"null\": null,\n\t\"number\": 42,\n\t\"object\": {},\n\t\"string\": \"Hello world\"\n}");
         }
 
         SECTION("indent=4")
         {
             CHECK(j.dump(4) ==
-                  "{\n    \"array\": [\n        1,\n        2,\n        3,\n        4\n    ],\n    \"boolean\": false,\n    \"null\": null,\n    \"number\": 42,\n    \"object\": {},\n    \"string\": \"Hello world\"\n}");
```

Comment:
You changed the previous dump output here. Please revert this.

Reply:
If the one-line array print is going to be accepted, existing tests for array print checks must also be changed. Honestly I am also afraid that changing default array print formats will incur backward compatibility issues. Maybe we should use a switch to control the one-line array format? By default the switch is off so that existing formats are maintained.

Reply:
The thing is: another switch to dump() won't be helping - it already has too many options. We rather need an approach for a second overload of dump() that takes a proper configuration. For now, I would rather split the support for multi-dimensional arrays (this PR) and their formatting (later PR).

Reply:
OK. I'll remove the dump formatting stuff from this PR. Thanks for the suggestion.